### DISCUSSION TITLE
Table support?

### DISCUSSION THREAD

**Discussion by cprieto** [apache/arrow-rs#3636] [Q&A] [answered]
I am more familiar with `pyarrow` and C++ arrow implementations and now learning or trying to grasp the same concepts in Rust but something that I had been struggling is with the `Table` type in Arrow (see https://arrow.apache.org/docs/python/generated/pyarrow.Table.html and https://arrow.apache.org/docs/cpp/tables.html). Looking around the crate documentation I cannot find anything about `Table` or similar mechanism. Am I wrong or missing something or we don't have `Table` in `arrow-rs`?

**Comment by tustvold** [comment]
I think https://docs.rs/arrow-array/latest/arrow_array/struct.RecordBatch.html is possibly what you are after?

**Reply by cprieto** [reply]
No, tables and record batches are different things (see the C++ documentation). You can construct a Table from record batches or from column arrays.

**Reply by alamb** [reply, marked as answer]
Looks to me like tables https://arrow.apache.org/docs/python/generated/pyarrow.Table.html contain very similar functionality to "data frames" with filtering, etc -- perhaps a library built on arrow-rs like datafusion might have what you are looking for:

https://docs.rs/datafusion/17.0.0/datafusion/dataframe/struct.DataFrame.html
